info-articles/Mapping in solidity - issue #19.md/article.md

```diff
@@ -0,0 +1,414 @@
+
+2. ### Tracking Allowance or Approvals
+In implementations such as ERC-20, addresses can allow or approve the address that owns the contract to send ether on their behalf. 
+```
+mapping(address => mapping(address => uint)) public allowance; 
+```
+The mapping above is a nested one that involves two sets of addresses where one allows or approves the other to take a specific action.
+3. ### Identifying owners
+There are instances when you will need to track the owners of some IDs in ERC-721. This is how to do it:
+```
+mapping(uint => address) internal _ownerOf;
+```
+Since we are more concerned about IDs, we will point uint to address.
+
+## How Mappings Are Stored in Solidity
+Mapping in itself is not capable of storing data. The only extent to which they are helpful in helping us tie some key types to value types. This is why you will discover that you cannot get the “.length” of mappings because they do not store actual data in real-time.
+
+But while it does not store data, we will need the hash of the key to trace its value pair. The hashes are stored in the storage slots of the contract. As a result, mappings can only declared at the state level and should be stored in storage; not calldata or memory. 
+
+There are a couple of reasons for this strict data location syntax. First, memory and calldata have a temporary nature, and are always declared locally. Therefore, they are unfit to be the data location of hashes that will be needed from time to time. 
+
+But in the case of storage, it. Thus, it is the fittest data location.
+
+## How to Set, Get, Update and Delete Mappings
+It is one thing to know how to create mappings, but it is another thing to be skillful with it while building your contract. You can set, get, update, and delete mappings in Solidity. Here is an example of such contracts:
+```
+// SPDX-License-Identifier : MIT
+ 
+ pragma solidity ^0.8.17;
+ 
+  contract MappingLesson{
+   
+    mapping(address => uint) public theBalanceOf;
+ 
+    function settingMapping() external{
+     theBalanceOf[msg.sender] = 3 ether;
+    }
+   
+    function gettingMapping() external view{
+      // for set address
+      uint bal = theBalanceOf[msg.sender];
+ 
+      // for unset address
+      uint bal13 = theBalanceOf[address(13)];
+    }
+ 
+    function updatingMapping() external{
+       theBalanceOf[msg.sender] += 13 ether;
+       // this will be the result of 13 + 3
+       // 16 total ethers
+    }
+ 
+    function clearingMapping() external{
+      delete theBalanceOf[address(13)];
+      delete theBalanceOf[msg.sender];
+    }
+ 
+ 
+  }
+```
+Now, let us break these processes down.
+
+### Setting Mapping
+Setting the value for mapping is quite straightforward; place the mapping beside an address that receives ether and assign it to a value.
+```
+function settingMapping() external{
+     theBalanceOf[msg.sender] = 3 ether;
+    }
+```
+### Getting Mapping

```

Comment:
Nice little section. I feel one more thing could be added. 
We don't have to always assign mapping value to a new variable to get a mapping. We can also simply return a mapping value from a function without using another variable.

Something like this-

**function gettingMapping2() external view returns(uint){
        return theBalanceOf[msg.sender];
    }**

Reply:
Added